clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -544,7 +547,8 @@ public void handle(JoinGroupResponse joinResponse, RequestFuture<ByteBuffer> fut
                 future.raise(error);
             } else if (error == Errors.INCONSISTENT_GROUP_PROTOCOL
                     || error == Errors.INVALID_SESSION_TIMEOUT
-                    || error == Errors.INVALID_GROUP_ID) {
+                    || error == Errors.INVALID_GROUP_ID
+                    || error == Errors.MEMBER_ID_MISMATCH) {
```

Comment:
If it's not too much work, maybe we could add a test to ensure `MEMBER_ID_MISMATCH` is fatal?

Reply:
Yea of course, will try to see how to make that happen